pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -87,13 +87,25 @@ func validateFloats(annotations map[string]string) *apis.FieldError {
 
 func validateWindows(annotations map[string]string) *apis.FieldError {
 	var errs *apis.FieldError
-	if v, ok := annotations[WindowAnnotationKey]; ok {
-		d, err := time.ParseDuration(v)
+	if w, ok := annotations[WindowAnnotationKey]; ok {
+		d, err := time.ParseDuration(w)
 		if err != nil {
-			errs = apis.ErrInvalidValue(v, WindowAnnotationKey)
+			errs = apis.ErrInvalidValue(w, WindowAnnotationKey)
 		} else if d < WindowMin || d > WindowMax {
-			errs = apis.ErrOutOfBoundsValue(v, WindowMin, WindowMax, WindowAnnotationKey)
+			errs = apis.ErrOutOfBoundsValue(w, WindowMin, WindowMax, WindowAnnotationKey)
 		}
+		if class, ok := annotations[ClassAnnotationKey]; ok {
+			switch class {
```

Comment:
In this case it might be simpler to just do `if class == HPA`. I also think this should move to the beginning of the function to bail out early and not hide the actual error with potential parsing issues, i.e.

```golang
if class := annotations[ClassAnnotationKey]; class == HPA {
  return errs.Also(apis.ErrInvalidKeyName(WindowAnnotationKey, class))
}

... // The current validation logic.
```

Reply:
Modified as per the suggestion.
Thank you